pkg/features/kube_features.go

```diff
@@ -2363,6 +2373,10 @@ var defaultKubernetesFeatureGateDependencies = map[featuregate.Feature][]feature
 
 	ResourceHealthStatus: {DynamicResourceAllocation},
 
+	// RestartAllContainersOnContainerExits introduces a new container restart rule action.
+	// All restart rules will be dropped by API if ContainerRestartRules feature is not enabled.
+	RestartAllContainersOnContainerExits: {ContainerRestartRules},
```

Comment:
You should add a dependency on NodeDeclaredFeatures, and add a feature for this. See https://github.com/kubernetes/kubernetes/pull/135203/files for an example.